.github/workflows/main.yml

```diff
@@ -34,6 +37,18 @@ jobs:
       - name: Create Version
         run: echo "VERSION=3.$(TZ=\"Europe/Oslo\" date +%Y.%m.%d_%H.%M)-$(git rev-parse --short=12 HEAD)" >> $GITHUB_ENV
 
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17
+          cache: 'maven'
+
+      - name: Deploy release
+        run: mvn -B -Drevision="${{ env.VERSION }}" -DskipTests source:jar deploy
```

Comment:
burde vi slå sammen stegene med tester og deploy?

Reply:
Tenker du at filen ser ut som dette? 
```
name: Test and release

on:
  push:
    branches:
      - main

jobs:
  test-and-release:
    name: Create new release
    permissions:
      contents: write
      packages: write
    runs-on: ubuntu-latest
    needs: test
    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Create Version
        run: echo "VERSION=3.$(TZ=\"Europe/Oslo\" date +%Y.%m.%d_%H.%M)-$(git rev-parse --short=12 HEAD)" >> $GITHUB_ENV

      - name: Set up JDK 17
        uses: actions/setup-java@v3
        with:
          distribution: temurin
          java-version: 17
          cache: 'maven'

      - name: Build, test and deploy release
        run: mvn -B -Drevision="${{ env.VERSION }}" source:jar deploy
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

      - name: Create release
        uses: softprops/action-gh-release@v1
        with:
          name: Release ${{ env.VERSION }}
          tag_name: ${{ env.VERSION }}
          prerelease: false
```

Reply:
Fordelen med å ha det delt opp er at hvis kun release-steget feiler så slipper vi å kjøre hele bygget fra start på nytt hvis det bare var noe midlertidig. Og så kan det være lettere å se hva som feiler. Men akkurat her er det jo så lite som skjer i teststeget at det kanskje ikke har så mye for seg?

Reply:
Jeg tror kanskje jeg vil beholde det som to steg, ved nærmere ettertanke, hvis ikke det er sterke meninger om å slå dem sammen. Vi er flinke til å låne hverandres kode her i NAV, og sånn det er så er dette et ganske rent og minimalt oppsett for å lage en maven-release med github package registry. Det kan være nyttig for nestemann som kommer og leter etter det samme 😊